x-pack/plugins/security/public/nav_control/nav_control_component.test.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { shallowWithIntl } from 'test_utils/enzyme_helpers';
+import { SecurityNavControl } from './nav_control_component';
+import { AuthenticatedUser } from '../../common/model';
+import { EuiPopover } from '@elastic/eui';
+
+describe('SecurityNavControl', () => {
```

Comment:
nit: we're missing tests for when the `EuiPopover` is displayed...

Reply:
Happy to add one. Out of curiosity, what benefits do we get by adding the test here that we don't get from the existing functional tests which verify the popover rendering? Is it mostly an argument for "defense in depth"?

Reply:
In this situation, we likely don't get that much benefit. However, in general I'd like to see us test less permutations in functional tests and start to rely on unit tests more, as they're much faster, less fragile, and easier to test more "permutations" including timing related situations.

Reply:
Makes sense. I think that's a general goal that we can all agree on.
To be clear, you don't want the functional tests removed in this case, right?

Reply:
Correct, we want as much high-level coverage using the functional tests as possible.